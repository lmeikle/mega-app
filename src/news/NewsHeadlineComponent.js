import React from 'react';
import { string } from 'prop-types';
import './News.css';

/**
 * Stateless component which renders a single news headline.
 */
const NewsHeadlineComponent = ({ title, date, source, url }) => {
  return (
    <div className="news-headline">
      <a href={url} target="_blank" rel="noopener noreferrer">
        <div className="news-headline-title">{title}</div>
      </a>
      <div className="news-headline-date-and-source-container">
        <div className="news-headline-date">{date}</div>
        <div className="news-headline-source">{source}</div>
      </div>
    </div>
  );
};

NewsHeadlineComponent.propTypes = {
  title: string.isRequired,
  date: string.isRequired,
  source: string.isRequired,
  url: string.isRequired
};

NewsHeadlineComponent.defaultProps = {};

export default NewsHeadlineComponent;
