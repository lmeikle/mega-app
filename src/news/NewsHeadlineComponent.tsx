import React from 'react';
import { IHeadLineProps } from './NewsContainer';
import './News.css';

/**
 * Stateless component which renders a single news headline.
 */
const NewsHeadlineComponent = ({ title, date, source, url }: IHeadLineProps) => {
  return (
    <div className="news-headline">
      <a href={url} target="_blank" rel="noopener noreferrer">
        <div data-testid="NewsHeadlineComponent_title" className="news-headline-title">
          {title}
        </div>
      </a>
      <div className="news-headline-date-and-source-container">
        <div className="news-headline-date">{date}</div>
        <div className="news-headline-source">{source}</div>
      </div>
    </div>
  );
};

export default NewsHeadlineComponent;
