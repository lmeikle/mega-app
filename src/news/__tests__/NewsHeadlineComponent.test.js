import React from 'react';
import { shallow } from 'enzyme';
import NewsHeadlineComponent from '../NewsHeadlineComponent';

let props = {
  title: 'Putin ready to meet Trump over arms race concerns',
  date: '10/06/2018',
  source: 'CNN',
  url: 'https://www.cnn.com/2018/06/10/politics/putin-trump-us-russia-arms-intl/index.html'
};

const newsHeadlineComponent = shallow(<NewsHeadlineComponent {...props} />);

test('renders correctly', () => {
  expect(newsHeadlineComponent).toMatchSnapshot();
});
