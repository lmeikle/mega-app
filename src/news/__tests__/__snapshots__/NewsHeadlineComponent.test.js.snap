// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewsHeadlineComponent
    date="10/06/2018"
    source="CNN"
    title="Putin ready to meet Trump over arms race concerns"
    url="https://www.cnn.com/2018/06/10/politics/putin-trump-us-russia-arms-intl/index.html"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <a
          href="https://www.cnn.com/2018/06/10/politics/putin-trump-us-russia-arms-intl/index.html"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            className="news-headline-title"
            data-testid="NewsHeadlineComponent_title"
          >
            Putin ready to meet Trump over arms race concerns
          </div>
        </a>,
        <div
          className="news-headline-date-and-source-container"
        >
          <div
            className="news-headline-date"
          >
            10/06/2018
          </div>
          <div
            className="news-headline-source"
          >
            CNN
          </div>
        </div>,
      ],
      "className": "news-headline",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="news-headline-title"
            data-testid="NewsHeadlineComponent_title"
          >
            Putin ready to meet Trump over arms race concerns
          </div>,
          "href": "https://www.cnn.com/2018/06/10/politics/putin-trump-us-russia-arms-intl/index.html",
          "rel": "noopener noreferrer",
          "target": "_blank",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Putin ready to meet Trump over arms race concerns",
            "className": "news-headline-title",
            "data-testid": "NewsHeadlineComponent_title",
          },
          "ref": null,
          "rendered": "Putin ready to meet Trump over arms race concerns",
          "type": "div",
        },
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="news-headline-date"
            >
              10/06/2018
            </div>,
            <div
              className="news-headline-source"
            >
              CNN
            </div>,
          ],
          "className": "news-headline-date-and-source-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "10/06/2018",
              "className": "news-headline-date",
            },
            "ref": null,
            "rendered": "10/06/2018",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "CNN",
              "className": "news-headline-source",
            },
            "ref": null,
            "rendered": "CNN",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            href="https://www.cnn.com/2018/06/10/politics/putin-trump-us-russia-arms-intl/index.html"
            rel="noopener noreferrer"
            target="_blank"
          >
            <div
              className="news-headline-title"
              data-testid="NewsHeadlineComponent_title"
            >
              Putin ready to meet Trump over arms race concerns
            </div>
          </a>,
          <div
            className="news-headline-date-and-source-container"
          >
            <div
              className="news-headline-date"
            >
              10/06/2018
            </div>
            <div
              className="news-headline-source"
            >
              CNN
            </div>
          </div>,
        ],
        "className": "news-headline",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="news-headline-title"
              data-testid="NewsHeadlineComponent_title"
            >
              Putin ready to meet Trump over arms race concerns
            </div>,
            "href": "https://www.cnn.com/2018/06/10/politics/putin-trump-us-russia-arms-intl/index.html",
            "rel": "noopener noreferrer",
            "target": "_blank",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Putin ready to meet Trump over arms race concerns",
              "className": "news-headline-title",
              "data-testid": "NewsHeadlineComponent_title",
            },
            "ref": null,
            "rendered": "Putin ready to meet Trump over arms race concerns",
            "type": "div",
          },
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="news-headline-date"
              >
                10/06/2018
              </div>,
              <div
                className="news-headline-source"
              >
                CNN
              </div>,
            ],
            "className": "news-headline-date-and-source-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "10/06/2018",
                "className": "news-headline-date",
              },
              "ref": null,
              "rendered": "10/06/2018",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "CNN",
                "className": "news-headline-source",
              },
              "ref": null,
              "rendered": "CNN",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
